src/CoreWCF.Primitives/src/CoreWCF/Security/WSSecurityOneDotZeroReceiveSecurityHeader.cs

```diff
@@ -125,8 +125,21 @@ protected override SignedXml ReadSignatureCore(XmlDictionaryReader signatureRead
                 writer.WriteEndDocument();
             }
             SignedXMLInternal signedXml = new SignedXMLInternal(doc);
-            XmlNodeList nodeList = doc.GetElementsByTagName("Signature");
+            XmlNodeList nodeList = doc.GetElementsByTagName("Signature", SignedXml.XmlDsigNamespaceUrl);
             signedXml.LoadXml((XmlElement)nodeList[0]);
+            if (signedXml.SignedInfo.CanonicalizationMethodObject is XmlDsigExcC14NTransform xmlDsigExcC14NTransform)
+            {
```

Comment:
I tried to do some validation looking at the behavior with our existing tests and it looks like we usually have XmlDsigC14NTransform as our transform so any problems here won't be picked up by our existing tests. The main thing I'm concerned with here is that signatureReader might not have a namespace matching the prefix when you do the lookup. Can you null check the value returned from LookupNamespace and only add if it isn't null.